docs/RFCS/information_schema.md

```diff
@@ -0,0 +1,216 @@
+these virtual tables.
+
+This RFC proposes that sometime after support for `information_schema` is added, we
+follow MySQL's lead and turn `SHOW` statements into a wrapper around `information_schema`
+queries. This will help eliminate duplicate implementations and simplify SHOW statement
+handling significantly.
+
+#### Schema Search Path Extension (only needed for `pg_catalog` support)
+
+PostgreSQL has a notion of a "Schema Search Path". In much the same way that we support the
+`SET database = xyz`, which will search for unqualified table names in the `xyz` database, the
+schema search path allows unqualified table names to be associated with a specific
+database/table pair. However, unlike our session database setting, PostgreSQL allows multiple
+databases to be in the search path. This is important because a large percent of the the time
+that `pg_catalog` tables are referenced in ORMs and other external tools, they are
+referenced using unqualified table names. For instance, a number of the ORMs use
+`pg_catalog` like:
+
+```sql
+SELECT * from pg_class
+```
+
+instead of
+
+```sql
+SELECT * from pg_catalog.pg_class
+```
+
+The reason this is allowed is because PostgreSQL includes `pg_catalog` in its
+database search path by default. In order to support this use case, we should extend our
+notion of a single database search path to a set of search paths. Like support for
+`pg_catalog`, this does not need to be completed during the initial implementation of
+this RFC.
+
+# Drawbacks
+
+- Complicates some SQL descriptor logic
+- Complicates database introspection on itself (ie. `SHOW TABLES FROM information_schema`)
+- Adds a few special cases to the query engine, but this is much less intrusive than
+  if we tried to mock out the database and table existence at a higher level than
+  descriptors
+
+# Alternatives
+
+- Create and maintain real tables just like we do with SystemDB tables
+- Mock out database and table existence at a higher level than [virtual
+  descriptors](#virtual-database-and-table-descriptors)
+- Begin building out VIEW infrastructure, and treat `information_schema` like
+  a read-only view. As noted in [Relation to VIEWs](#relation-to-views), this
+  wouldn't actually change very much, and would still require a lot of
+  this work. For instance, we would still need to somehow mock out the existence
+  of the `information_schema` database itself, and would still need to have the
+  VIEWs source their own introspective data.
+
+# Unresolved questions
+
+### CockroachDB Extensions
+
+This RFC proposed [above](#standards-considerations) that we could add relevant extension
+columns to our `information_schema` tables, as long as they do not conflict with other columns,
+and as long as they add concrete value. An example here would be a column in the
+`INFORMATION_SCHEMA.TABLES` table about column families. What other columns or tables would be
+valuable CockroachDB-extensions to the `information_schema`?
+
```

Comment:
Could it? The `information_schema.tables` table will be populated with a scan similar to https://github.com/cockroachdb/cockroach/blob/3dbb3cb581cc73a83b72929109e4ee4f14d64710/sql/table.go#L304, which should avoid issues like that.